src/flist/db.rs

```diff
@@ -0,0 +1,120 @@
+use anyhow::{bail, Result};
+use bytes::Buf;
+use std::io;
+use std::path::{Path, PathBuf};
+use tokio::fs;
+
+pub struct MetadataDbMgr {
+    // root directory where all
+    // the working file of the module will be located
+
+    // underneath are the path for each
+    // sub folder used by the flist module
+    flist: PathBuf,
+}
+
+impl MetadataDbMgr {
+    pub async fn new<P: AsRef<Path>>(flist: P) -> Result<Self> {
+        Ok(Self {
+            flist: flist.as_ref().to_path_buf(),
+        })
+    }
+    pub async fn get<T: AsRef<str>>(&self, url: T) -> Result<PathBuf> {
+        let url = url.as_ref();
+        let hash = self.hash_of_flist(url).await?;
+        let path = self.flist.join(&hash.trim());
```

Comment:
according to code of `hash_of_flist()` you already do the trimming. why call trim here again. you should trust that your functions return always correct data or error